### DISCUSSION TITLE
Updated Proxmox VE LXC Updater

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#1764] [Announcements]
We've added a new menu that allows you to choose which containers you want to exclude from the update process. This feature provides greater control over the update procedure. We're eager to hear everyone's feedback on this addition and gather their thoughts.

**Comment by tteck** [comment]
Even though the new menu displays the Container IDs of templates, it continues to automatically skip them during the update process.

**Reply by towerhand** [reply]
Awesome. It works like a charm. Would it make sense to add the same to the Proxmox VE LXC Cleaner script?

**Reply by tteck** [reply]
- **Proxmox VE LXC Cleaner**
  - Added a new menu that allows you to choose which containers you want to exclude from the clean process.

**Comment by CoreyJ87** [comment]
Does this script update the actual applications or does it just do like apt/yum update/upgrade?

**Reply by tteck** [reply]
Only the OS.

**Reply by timdonovanuk** [reply]
I'm wondering how other people are managing this. Other than manually updating each LXC every few weeks.

**Reply by tteck** [reply]
crontab?
https://github.com/tteck/Proxmox/blob/main/misc/cron-update-lxcs.sh

**Reply by timdonovanuk** [reply]
Yeah I meant for the application though. Does each lxc itself contain link or reference to the original script that created it, or will each lxc need a custom crontab? 

Thanks :)

**Reply by tteck** [reply]
As of Oct 5, 2023 each LXC created has an "update" command to run a script for updating the application. However, this approach is not unattended.

**Reply by ajtatum** [reply]
Hi @tteck, thanks for these scripts. I have the qBittorrent LXC and when I SSH into the machine and type "update" it says that the command is not found. Do I need to be in a specific directory? I did create the LXC before Oct 5, 2023 but have run the LXC update script for it. Would the update command be installed through the LXC update if it wasn't already installed?

**Reply by tteck** [reply]
The update command is only for LXCs created after Oct 5, 2023. You can continue to update via the script.

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/discussions/1900

**Comment by tismofied** [comment]
Is there a way to get notified (email or ntfy ,etc...) when lxcs gets updated using this script?

**Reply by tteck** [reply]
No. It's not an unattended process, so, why the need to be notified?

**Reply by AlexGeek77** [reply]
Getting notified when there's an update available for an installed LXC would be nice but might require too much work to maintain.

**Reply by tteck** [reply]
Have you tried Webmin?

**Reply by AlexGeek77** [reply]
> Have you tried Webmin?

It's been a while since I had Webmin running. I see there's a Software Packages module for it. I'll check it out. Thanks.

**Reply by tismofied** [reply]
> No. It's not an unattended process, so, why the need to be notified?

I must have misunderstood the way the script work with crontab scheduling. I thought the script would run every Sunday at midnight to run the updater script against all LXCs.

**Reply by tteck** [reply]
That would be the **Proxmox VE Cron LXC Updater**, it is unattended. Maybe I just misunderstood.

**Reply by tismofied** [reply]
Yeah that's what I was talking about lol. 
So... is it feasible ?

**Reply by tteck** [reply]
Well, the Proxmox VE Cron LXC Updater script would need to be modified to log its output, then the log could be emailed, if the command-line utility called `ssmtp` is installed and set up correctly. Or, possibly sending emails could be set up using Webmin. 🤷🏻‍♂️ 
Modifying the script to log its output I can do, but sending the email part, I have no idea as I've never done that.

**Reply by tismofied** [reply]
Thank you for all you do by the way. It's really not a big deal. I can live without. 😄
